core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -237,6 +255,9 @@ private[data] sealed trait NonEmptyVectorInstances {
       override def traverse[G[_], A, B](fa: NonEmptyVector[A])(f: (A) => G[B])(implicit G: Applicative[G]): G[NonEmptyVector[B]] =
         G.map2Eval(f(fa.head), Always(Traverse[Vector].traverse(fa.tail)(f)))(NonEmptyVector(_, _)).value
 
+      override def zipWithIndex[A](fa: NonEmptyVector[A]): NonEmptyVector[(A, Int)] =
+        fa.zipWithIndex
```

Comment:
this is untested, so I guess that means zipWithIndex is currently lawless.

Reply:
We could write a test to check if `zipWithIndex` is consistent with `zipWith(Range(0, size))` :)